src/Storages/DataLakes/IStorageDataLake.h

```diff
@@ -94,10 +104,11 @@ class IStorageDataLake : public Storage
         const bool updated = base_configuration.update(local_context);
         auto new_keys = getDataFiles(base_configuration, local_context);
 
-        if (!updated && new_keys == Storage::getConfiguration().keys)
+        if (!updated && !base_configuration.is_broken && new_keys == Storage::getConfiguration().keys)
```

Comment:
Why do we need `is_broken` flag? If we failed to parse metadata on storage creation and managed to do it on read in `getDataFiles` called above, we will have different `new_keys` and `Storage::getConfiguration().keys` and this condition will not pass. If we failed to parse metadata on read too, we will have an exception during `getDataFiles` call above.